### PR TITLE
Add Node Name to the output of nomad job status (verbose only) and nomad alloc status

### LINKED ISSUE #1180: Show node names instead of node ids in nomad status
It will be cool if in nomad status <app> command show node names, now it show node is and it's not very convenient

### LINKED ISSUE #2359: Implement CLI/API feature showing actual hostnames where allocation runs.
Hi, this is a feature request.

Since killing a specific allocation which is part of the same job with multiple instances is not implemented yet  (https://github.com/hashicorp/nomad/issues/2181)   it would be good for purposes of chasing down that bad behaving allocation and killing it manually with kill -9 <pid>  to add ability to both "nomad status <job id> " CLI  and API to show the actual hostname where allocation runs along with other info like Node ID.

### PR DESCRIPTION
**Pull request by arshjohar**
Building on the work done by @jrasell in #4516 .

Closes #2359
Closes #1180
Closes #4516 

The plan is to merge this into 0.9.1-dev, which I just checked out from the tip of master at this point in time. Changes to 0.9.1-dev will be merged into master, only after 0.9.0 is released.

Relevant snippets from the output:
```shell
➜  nomad git:(jrasell-gh_2359) nomad job status -verbose example
Allocations
ID                                    Eval ID                               Node ID                               Node Name                     Task Group  Version  Desired  Status   Created                    Modified
f1cd3043-eb56-d533-5d9b-7e93b66195b6  4f2feeb6-ba25-b451-11eb-168f0b49102a  fc3a8aa9-95b2-1c16-5ab0-f53be2d887fa  Arshneets-MBP.hitronhub.home  cache       0        run      running  2019-01-22T13:52:30-08:00  2019-01-22T13:52:31-08:00
```

```shell
➜  nomad git:(jrasell-gh_2359) nomad alloc status f1cd3043  
ID                  = f1cd3043
Eval ID             = 4f2feeb6
Name                = example.cache[0]
Node ID             = fc3a8aa9
Node Name           = Arshneets-MBP.hitronhub.home
Job ID              = example
Job Version         = 0
Client Status       = running
Client Description  = Tasks are running
Desired Status      = run
Desired Description = <none>
Created             = 2m11s ago
Modified            = 2m10s ago
Deployment ID       = 954801b1
Deployment Health   = unset
```

### CODE REVIEW

File: command/alloc_status_test.go [hunk 1]
```diff
@@ -141,6 +143,16 @@ func TestAllocStatusCommand_Run(t *testing.T) {
 		t.Fatalf("expected to have 'Modified' but saw: %s", out)
 	}
 
+	if !strings.Contains(out, "Modified") {
```
Comment (dantoml) [inline command/alloc_status_test.go]: Accidental duplicate?

File: command/alloc_status_test.go [hunk 2]
```diff
@@ -141,6 +143,16 @@ func TestAllocStatusCommand_Run(t *testing.T) {
...
+		t.Fatalf("expected to have 'Modified' but saw: %s", out)
+	}
+
+	nodeNameRegexpStr := fmt.Sprintf(`\nNode Name\s+= %s\n`, regexp.QuoteMeta(nodeName))
```
Comment (dantoml) [inline command/alloc_status_test.go]: If you migrate this test to using `require`, you can use `require.Regexp(regexp.MustCompile(regexpStr), out)`, which provides a nicer error message here. (example in this test: https://github.com/hashicorp/nomad/blob/59dec15a7a0ec8d0cdf59e5c30d7292281c51be0/command/alloc_status_test.go#L236-L237)

File: command/alloc_status_test.go (hunk 1 above)
Comment (arshjohar) [reply command/alloc_status_test.go]: Yes, thanks for pointing it out.

File: command/alloc_status_test.go (hunk 2 above)
Comment (arshjohar) [reply command/alloc_status_test.go]: Updated

**Pull request merged**